lib/adrian/directory_queue.rb

```diff
@@ -69,9 +72,10 @@ def reserve(item)
     end
 
     def items
-      items = files.map { |file| wrap_item(file) }
-      items.reject! { |item| !item.exist? || filter?(item) }
-      items.sort_by(&:updated_at)
+      return @items if @items && @items.length > 0
+      @items = files.map! { |file| wrap_item(file) }
+      @items.reject! { |item| !item.exist? || filter?(item) }
```

Comment:
nevermind then :)

On Thu, May 21, 2015 at 11:22 AM, Rebecca Martz <notifications@github.com>
wrote:

> In lib/adrian/directory_queue.rb
> <https://github.com/zendesk/adrian/pull/3#discussion_r30831315>:
>
> > @@ -69,9 +72,10 @@ def reserve(item)
> >      end
> >
> >      def items
> > -      items = files.map { |file| wrap_item(file) }
> > -      items.reject! { |item| !item.exist? || filter?(item) }
> > -      items.sort_by(&:updated_at)
> > +      return @items if @items && @items.length > 0
> > +      @items = files.map! { |file| wrap_item(file) }
> > +      @items.reject! { |item| !item.exist? || filter?(item) }
>
> Not super easily because filter? can be used on a list of QueueItems
> rather than FileItems and QueueItem does not have an exist? method. We know
> we have FileItems here because we've "wrapped" them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/adrian/pull/3/files#r30831315>.
>